### ISSUE TITLE
Fix S1144 FP: Unused private types or members should be removed -> Does not recognize usages of private members in razor files / partial Blazor components

### ISSUE THREAD

**Issue by naefp** [SonarSource/sonar-dotnet#8537]
### Description

When declaring private member or method in a razor.cs file, but only accessing it in the related razor file, sonar marks it as not used. Sonar should check the related razor file for [S1144](https://sonarsource.github.io/rspec/#/rspec/S1144/csharp) as well.

### Repro steps

```html
<Button Clicked="@OnButtonOkClicked">
  OK
</Button>
```

```c#
private Task OnButtonOkClicked()  // Sonar marks this method as not used
{
    // Do something
}
```

### Expected behavior

Recognize that the method actually is being used in razor file.

### Actual behavior

False positive for [S1144](https://sonarsource.github.io/rspec/#/rspec/S1144/csharp).

### Known workarounds

Mark every occurrence as false positive, or disable rule `csharpsquid:S1144` for `**/*.razor.cs` completely.

Probably not using code in separate *razor.cs file but declare the private field in the razor file itself. Did not test that.

### Related information

* C# 12
* Rider 2023.3.2
* .Net 8
* SonarLint plugin 10.2.1.77304 connected to SonarCloud
* Windows 11

**Comment by cristian-ambrosini-sonarsource** [comment]
Hi @naefp! Quick question, is the rule raising on SonarCloud, just in the IDE, or both?

**Comment by costin-zaharia-sonarsource** [comment]
I reproduced this using JetBrains Rider 2023.3.2

**Comment by cristian-ambrosini-sonarsource** [comment]
I confirm this as a FP. To provide additional context, the issue does not reproduce in the following:
- SL4VS
- local analysis run with SonarQube 
- analyzer UTs 
Furthermore, this issue may be related to #8535, as they both exhibit similar behavior and share the same implementation (see [here](https://github.com/SonarSource/sonar-dotnet/blob/4383deee2e25a3830993e5436509dc81afc72d9f/analyzers/src/SonarAnalyzer.CSharp/Rules/UnusedPrivateMember.cs#L28)).

**Comment by sebastien-marichal** [comment]
The issue seems to be related to SonarLint only.
Since it is not yet able to analyze Razor generated files, it is not able to see the method usage in the component.
We have an internal ticket to tackle the support of Razor in SonarLint which should fix this issue.

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1629](https://sonarsource.atlassian.net/browse/NET-1629)

[NET-1629]: https://sonarsource.atlassian.net/browse/NET-1629?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
